Add Sentry.metrics for React Native SDK (#9119)
getsentry/sentry-docs: docs/platforms/react-native/metrics/index.mdx

## Configure Metrics

<PlatformContent includePath="metrics/configure" />

## Emit a Counter

Counters are one of the more basic types of metrics and can be used to count certain event occurrences.

To emit a counter, do the following:

<PlatformContent includePath="metrics/counter-snippet" />

## Emit a Distribution

Distributions help you get the most insights from your data by allowing you to obtain aggregations such as `p90`, `min`, `max`, and `avg`.

To emit a distribution, do the following:

<PlatformContent includePath="metrics/distribution-snippet" />

## Emit a Set

Sets are useful for looking at unique occurrences and counting the unique elements you added.

To emit a set, do the following:

<PlatformContent includePath="metrics/set-snippet" />

## Emit a Gauge

Gauges let you obtain aggregates like `min`, `max`, `avg`, `sum`, and `count`. They can be represented in a more space-efficient way than distributions, but they can't be used to get percentiles. If percentiles aren't important to you, we recommend using gauges.

To emit a gauge, do the following:

<PlatformContent includePath="metrics/gauge-snippet" />

## Supported Performance Metric Units

> You can pass a unit as a optional parameter of the third argument of the `increment`, `distribution`, `set`, and `gauge` methods.
> 
> Units augment metric values by giving meaning to what otherwise might be abstract numbers. Adding units also allows Sentry to offer controls - unit conversions, filters, and so on - based on those units. For values that are unitless, you can supply an empty string or `none`.
> 
> These following units are understood by the Sentry backend, but you can supply any arbitrary units you want.

Reviewer: ```suggestion
Adding a unit as a metric parameter will give meaning to what may otherwise look like abstract numbers. It also allows Sentry to offer controls - unit conversions, filters, and so on - based on the unit you select. You can pass a unit as an optional parameter of the third argument of the `increment`, `distribution`, `set`, and `gauge` methods. If a value has no units, you can supply an empty string or `none`.

The following units are understood by the Sentry backend, but you can add any arbitrary unit you want.
```

### Duration Units

- `nanosecond`
- `microsecond`
- `millisecond`
- `second`
- `minute`
- `hour`
- `day`
- `week`

### Information Units

- `bit`
- `byte`
- `kilobyte`
- `kibibyte`
- `megabyte`
- `mebibyte`
- `gigabyte`
- `gibibyte`
- `terabyte`
- `tebibyte`
- `petabyte`
- `pebibyte`
- `exabyte`
- `exbibyte`

### Fraction Units

- `ratio`
- `percent`

> If you want to explore further, you can find details about supported units in our [event ingestion documentation](https://getsentry.github.io/relay/relay_metrics/enum.MetricUnit.html).

Reviewer: ```suggestion
You can find additional details about supported units in our [event ingestion documentation](https://getsentry.github.io/relay/relay_metrics/enum.MetricUnit.html).
```
